Sources/client/Socket.swift

```diff
@@ -147,51 +150,37 @@ public class Socket {
   //----------------------------------------------------------------------
   // MARK: - Initialization
   //----------------------------------------------------------------------
+  public convenience init(_ endPoint: String) {
+    self.init(endPoint: endPoint,
+              transport: { url in return WebSocket(url: url) },
+              params: { nil })
+  }
+
   public convenience init(_ endPoint: String,
-                          params: [String: Any]? = nil) {
+                          params: Payload?) {
```

Comment:
I would assigned `= nil` on this initializer and then not provide a default one for `paramsGetter`